### PR TITLE
[OASIS-12] Move pkgagent initialization to newAgent

### PR DESCRIPTION
**Pull request by songy23**
### What does this PR do?

Move pkgagent initialization from `start` to `newAgent` in trace agent component.

### Motivation

Make sure `newAgent` returns a concrete `pkgagent.Agent`

### Describe how to test/QA your changes

Deploy the new trace agent binary to staging cluster. Verify there are no race conditions and traces are still sent successfully.

### CODE REVIEW

File: comp/trace/agent/agent.go [hunk 1]
```diff
@@ -98,6 +98,17 @@ func newAgent(deps dependencies) Component {
 		tagger:             deps.Tagger,
 		wg:                 sync.WaitGroup{},
 	}
+	statsdCl, err := setupMetrics(ag.statsd, ag.config, ag.telemetryCollector)
```
Comment (knusbaum) [inline comp/trace/agent/agent.go:101]: Nit:
I don't think we need `statsd` in the `agent` struct anymore now that it's not being used anywhere but here.

**Review by knusbaum — CHANGES_REQUESTED** [review]
Minor nit. Otherwise LGTM.

File: comp/trace/agent/agent.go (hunk 1 above)
Comment (songy23) [reply comp/trace/agent/agent.go:101]: `pkgagent.NewAgent` still needs a statsd client. Do you mean using a noop client here?

Comment (songy23) [reply comp/trace/agent/agent.go]: NVM, done in https://github.com/DataDog/datadog-agent/pull/26105/commits/6ee81d4a3fadf1f712c5edfc02df97b05572a10d

**Review by dineshg13 — APPROVED** [review]

**Review by knusbaum — APPROVED** [review]
lgtm

**Pull request merged**